### DISCUSSION TITLE
Ability to update variable value in collection, environment and global file

### DISCUSSION THREAD

**Discussion by cheungm1** [postmanlabs/postman-flows#178] [Ideas]
I got a API flow to run which needs a unique projectName to start, I use a random number generator to create the projectName and save it to the collection variable, as the Flow runner cannot save value to the collection variable, so I need to change the projectName manually each time before I run the Flow. Therefore, if the Flow has the function to save any value back to any variable in the collection file will be much helpful.

**Comment by KMiszczak91** [comment]
Saving values from a response into a collection would be great! Just need this right now.
I don't understand why I can select an environment in a Send Request block if I cannot use the env variables in there

**Comment by tomalazarov** [comment]
> @cheungm1 I had the same issue but found a solution. My case is a little bit different but the idea is the same... I had to generate something randomly. Let's say in my case it's random account number. So yes, it's easy to make it a collection variable and stuff, we all know that. But flows does not use collection variables so this is what I've tried and it works :DD - Create Durable where the Key is the name you are going to use in the request (in my case it's randomAcc) and the Value is a string "{{$timestamp}}" (I just use it for some random nums but ofc you can use whatever it suits you from the Dynamic Variables page https://learning.postman.com/docs/writing-scripts/script-references/variables-list/ ).
<img width="789" alt="Screenshot 2022-01-06 at 14 20 11" src="https://user-images.githubusercontent.com/48013428/148382713-d3ff0621-6523-46a8-9f6f-060004c4e81f.png">
<img width="878" alt="Screenshot 2022-01-06 at 14 24 20" src="https://user-images.githubusercontent.com/48013428/148382719-f60c238d-35a9-49cb-9f48-a64582acd118.png">
<img width="876" alt="Screenshot 2022-01-06 at 14 34 25" src="https://user-images.githubusercontent.com/48013428/148384338-12ed7d45-bd5e-4684-abbb-e6d711c30912.png">
